### PR TITLE
[FEATURE] CLIP (DiffusionDB2M finetune, InfoNCE, Callback)

### PR DESCRIPTION
**Pull request by sami-bg**
## Description

Add InfoNCELoss: CLIP-style symmetric contrastive loss with temperature / logit scale control and DDP support.

Add CLIPZeroShot callback: evaluates zero-shot classification by comparing image embeddings to class text embeddings, logging torchmetrics during validation.

Unit tests for InfoNCELoss and ClipZeroShot

Finetuning benchmark script for DiffusionDB2M experiment
Eval benchmark script for DiffusionDB2M on ImageNet100

## Checklist

- [x] I have read the [**Contributing**](https://rbalestr-lab.github.io/stable-SSL.github.io/dev/contributing.html#) document.
- [x] The documentation is up-to-date with the changes I made (check build artifacts).
- [x] All tests passed, and additional code has been **covered with new tests**.
- [ ] I have added the PR to the [**RELEASES.rst**](../RELEASES.rst) file.

### CODE REVIEW

File: stable_pretraining/losses.py [hunk 1]
```diff
@@ -226,3 +226,61 @@ def forward(self, z_i, z_j):
...
+            text_feats: Text embeddings for the batch (paired with images).
+            logit_scale: Optional scalar (float or 0-dim tensor). If provided,
+                logits = logit_scale * (img @ txt^T). If None, uses
+                1/temperature.
+
+        Returns:
+            torch.Tensor: Scalar loss value.
+        """
+        # Gather across devices (DDP) to form global batch
+        img = torch.cat(all_gather(F.normalize(image_feats, dim=-1)), dim=0)
```
Comment (RandallBalestriero) [inline stable_pretraining/losses.py:265]: any way to use the lightning module method for gather? that would be safer, just need to make sure to give sync_grads=True since we use that for training

File: stable_pretraining/losses.py [hunk 2]
```diff
@@ -226,3 +226,61 @@ def forward(self, z_i, z_j):
         off_diag = off_diagonal(c).pow(2).sum()
         loss = on_diag + self.lambd * off_diag
         return loss
+
+
+class InfoNCELoss(torch.nn.Module):
```
Comment (RandallBalestriero) [inline stable_pretraining/losses.py:231]: if we can use the same loss for SimCLR that would be great (make sure it is exactly the same implementation) and in case case I would make the argument names more general (not image/text)

**Review by huguesva — APPROVED** [review]
